srml/council/src/lib.rs

```diff
@@ -14,7 +14,188 @@
+//! dangerous or malicious referenda.
+//!
+//! ### Terminology
+//!
+//! #### Council Proposals (motions.rs)
+//!
+//! - **Council motion:** A mechanism used to enact a proposal.
+//! - **Proposal validity:** A council proposal is valid when it's unique, hasn't yet been vetoed, and
+//! when the council term doesn't expire before the block number when the proposal's voting period ends.
+//! - **Proposal postponement:** Councillors that abstain from voting may postpone a council proposal from
+//! being approved or rejected. Postponement is equivalent to a veto, which only lasts for the cooloff period.
+//! - **Cooloff period:** Period, in blocks, for which a veto is in effect.
+//!
+//! #### Council Proposal Voting (voting.rs)
+//!
+//! - **Proposal:** A submission by a councillor. An initial vote of yay from that councillor is applied.
+//! Unlike the Democracy and Treasury modules, the `Proposal` type is generic and can be anything that
+//! can be dispatched (see [`type Proposal`](./motions/trait.Trait.html#associatedtype.Proposal)).
+//! - **Referendum:** The means of voting on a proposal.
+//! - **Vote:** A vote of yay or nay from a councillor on a single proposal. Councillors may change their vote.
+//! - **Veto:** A council member may veto any council proposal that exists. A vetoed proposal that's valid is set
+//! aside for a cooloff period. The vetoer cannot re-veto or propose the proposal again until the veto expires.
+//! - **Elevation:** A referendum can be _elevated_ from the council to a public referendum. This means it has
+//! been passed to the Democracy module for a public vote.
+//! - **Referendum cancellation:** At the end of a given block we cancel all elevated referenda whose voting period
+//! ends at that block and where the outcome of the vote tally was a unanimous vote to cancel the referendum.
+//! - **Voting process to elevate a proposal:** At the end of a given block we tally votes for expiring referenda.
+//! Referenda that are passed (yay votes are greater than nay votes plus abstainers) are sent to the Democracy
+//! module for a public referendum. If there are no nay votes (abstention is acceptable), then the proposal is
+//! for immediate enactment. Otherwise, there will be a delay period. If the vote is unanimous, then the public
+//! referendum will require a vote threshold of supermajority against to prevent it. Otherwise,
+//! it is a simple majority vote.
+//!
+//! #### Council Seats (seats.rs)
+//!
+//! - **Desired seats:** The number of seats on the council. Can change via governance.
+//! - **Candidacy bond:** Bond required to be a candidate.
+//! - **Leaderboard:** A list of candidates and their respective votes in an election, ordered low to high.
+//! - **Presentation:** The act of proposing a candidate for insertion into the leaderboard. Presenting is
+//! `O(number of voters)`, so the presenter must be slashable and will be slashed for duplicate or invalid
```

Comment:
Is there a way to optimize this?

Reply:
There is a more detailed explanation with the [code](https://github.com/paritytech/substrate/blob/913c726cfb3b39163ea84654a2d6e56007f9722f/srml/council/src/seats.rs#L29). Would be nice to optimize but also beyond the scope of the documentation. The account presenting is slashed heavily for abusing this, so although it's not optimal, it's not an attack vector.

Reply:
@DemiMarie-temp @DemiMarie-parity this might be relevant and interesting  https://github.com/paritytech/substrate/pull/2407